### PR TITLE
Welcome message

### PR DESCRIPTION
**Pull request by ismailsunni**
### What does it fix?
* Ticket: Part of #4482 
* Funded by: DFAT
* Description: 
   - Adding welcome message when you run InaSAFE for the first time.
   - It will check the setting for `inasafe/first_run`. If it's True, or not exist, it will show the welcome message. Currently, you can only get the message once. If you want to see it again, set `inasafe/first_run` setting in QGIS, and set it to True.

welcome_message

Of course I need you @timlinux to make a better message :D  

### Checklist:
- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Add to the changelog in metadata.txt if it's a new feature
- [x] Unit test for new code added
- [x] Request someone to review or test your PR

### CODE REVIEW

File: safe/gui/tools/help/welcome_message.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    return message
+
+
+def heading():
+    """Helper method that returns just the header.
+
+    This method was added so that the text could be reused in the
+    other contexts.
+
+    .. versionadded:: 4.1.0
```
Comment (Gustry) [inline safe/gui/tools/help/welcome_message.py]: 4.3 ?

File: safe/gui/tools/help/welcome_message.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    return message
+
+
+def content():
+    """Helper method that returns just the content.
+
+    This method was added so that the text could be reused in the
+    dock_help module.
+
+    .. versionadded:: 4.1.0
```
Comment (Gustry) [inline safe/gui/tools/help/welcome_message.py]: 4.3 ?

File: safe/gui/tools/help/welcome_message.py (hunk 1 above)
Comment (ismailsunni) [reply safe/gui/tools/help/welcome_message.py]: Ah yes, forget to update it. Thanks.

File: safe/gui/tools/help/welcome_message.py (hunk 2 above)
Comment (ismailsunni) [reply safe/gui/tools/help/welcome_message.py]: Same as above.

**Comment by ismailsunni** [comment]
After checking again, I am still not sure if my code is right. There are some case:
1. Show it after the plugin loaded / installed (call the function in initGui method)
2. Show after the QGIS is loaded (make a slot for `initializationCompleted` completed)

But, the first one will make the welcome message shown before the initializationCompleted (if we want to show it every time we open in QGIS).

**Comment by Gustry** [comment]
What is wrong with the initGUI ? Is-it not enough?

But please, we mustn't be too intrusive on UX. I hate plugins doing too much display or popup on launch time if I don't explicitly use them or call them

**Comment by ismailsunni** [comment]
A little update from the UI. Adding check box to always show it.
screen shot 2017-10-30 at 21 02 47
screen shot 2017-10-30 at 21 03 19

**Comment by ismailsunni** [comment]
> What is wrong with the initGUI ? Is-it not enough?

initGUI is finished before QGIS initialization is completed. So, it will be shown before QGIS (which is not good, I think). But all is good now, since we decide to show it only after open QGIS (not after load InaSAFE)

> But please, we mustn't be too intrusive on UX. I hate plugins doing too much display or popup on launch time if I don't explicitly use them or call them

Yeah, me too. We only show this if:
1. The user set always show the welcome message check box
2. If the user use newer version of InaSAFE (compared the previous one). And this only once per new version installed.

**Pull request merged**